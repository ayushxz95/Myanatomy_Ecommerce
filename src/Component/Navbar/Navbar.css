.nav{
  position: relative;
  z-index: 20;
}
@media (min-width: 800px) {
ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  overflow: hidden;
  background-color: #333;
  position: fixed;
  top: 0;
  width: 100%;
}
li {
  float: left;
}

li a {
  display: block;
  color: white;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
}
.active {
  color: #ef257a;
}
li a:hover{
  background-color: #111;
  cursor: pointer;
}
}
@media (max-width: 800px) {
  ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
    overflow: hidden;
    background-color: #333;
    position: fixed;
    top: 0;
    width: 100%;
  }
  li {
    float: left;
  }
  
  li a {
    display: none;
    color: white;
    text-align: center;
    padding: 14px 16px;
    text-decoration: none;
  }
  .active {
    color: #ef257a;
  }
  li a:hover{
    background-color: #111;
    cursor: pointer;
  }
  }
@media (max-width: 800px){
  .hamberger-menu-container{
    display: flex;
  }
}
@media (min-width: 800px) {
  .hamberger-menu-container{
    display: none;
  }
}
/* .Add-item{
  display: block;
  color: white;
  text-align: center;
  padding: 14px 18px;
  text-decoration: none;
}
.Add-item:hover {
  background-color: #111;
  cursor: pointer;
} */
@media (max-width: 800px) {
  

    .search-bar{
      margin-top: 8px;
      margin-left: 26px;
      width: 300px;
      height:10px;
      border-radius: 30px;
      display:none;
      flex-direction: row;
      background-color: white;
      padding: 2px;
      float: right;
      margin-right: 38px;
    }
  .search-bar > i{
    padding: 2px;
    font-size: 20px;
    color: #777;
  }
  .search-bar > input{
    flex:1;
    border:none;
    outline: none;
  }

}
@media (min-width: 800px) {
.search-bar{
  margin-top: 8px;
  margin-left: 26px;
  width: 300px;
  height:30px;
  border-radius: 30px;
  display:flex;
  flex-direction: row;
  background-color: white;
  padding: 4px;
  float: right;
  margin-right: 38px;
}
.search-bar > i{
  padding: 2px;
  font-size: 20px;
  color: #777;
}
.search-bar > input{
  flex:1;
  border:none;
  outline: none;
}
}