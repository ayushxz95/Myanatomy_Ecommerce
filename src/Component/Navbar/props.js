import {
    fetchCategoryItemsSuccess,
    fetchItems,
    searchItem
} from "../../Redux/Actions/ItemAction";
import {
  ChangeCategoryItem
} from "../../Redux/Actions/categoryAction";


const mapStateToProps = (state) => {
  return {
     category: state.category.category,
}
}

const mapDispatchToProps = (dispatch) => {
    return {
      fetchCategoryItems: (category) =>
        dispatch(fetchCategoryItemsSuccess(category)),
      fetchItems: () => dispatch(fetchItems()),
      searchItem: (text, category) => dispatch(searchItem(text, category)),
      changeCategory: (category) => dispatch(ChangeCategoryItem(category)),
    };
  };

  export {mapStateToProps, mapDispatchToProps};