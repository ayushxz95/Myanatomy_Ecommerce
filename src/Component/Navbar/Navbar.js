import React, { Component } from "react";
// import { withRouter } from "react-router";
import {
  fetchCategoryItemsSuccess,
  fetchItems,
} from "../../Redux/Actions/ItemAction";
import { connect } from "react-redux";
import "./Navbar.css";
import { GoSearch } from "react-icons/go";
import { PropTypes } from "prop-types";
import { searchItem } from "../../Redux/Actions/ItemAction";
import ResNavbar from "../ResNavbar/index";
import { toHaveDisplayValue } from "@testing-library/jest-dom/dist/matchers";


class Navbar extends Component {
  constructor(props) {
    super();
    this.state = {
      showModal: false,
      activeCategory: "Home",
    };
  }
  openModelClickHandler = () => {
    this.setState({
      showModal: !this.state.showModal,
    });
  };
  onChange = (e) => {
    this.props.searchItem(e.target.value, this.props.category);
  };
  render() {
    return (
      <div className="nav">
        <ul>
          <li className="hamberger-menu-container">
            <ResNavbar/>
          </li>
          {this.state.activeCategory === "Home" ? (
            <li>
              <a
                className="active"
                onClick={() => {
                  this.props.fetchItems();
                  this.setState({ activeCategory: "Home" });
                }}
              >
                Home
              </a>
            </li>
          ) : (
            <li>
              <a
                onClick={() => {
                  this.props.fetchItems();
                  this.props.changeCategory("Home");
                  this.setState({ activeCategory: "Home" });
                }}
              >
                Home
              </a>
            </li>
          )}

          {this.state.activeCategory === "Cloths" ? (
            <li>
              <a
                className="active"
                onClick={() => {
                  this.props.fetchCategoryItems("Cloths");
                  this.setState({ activeCategory: "Cloths" });
                }}
              >
                Cloths
              </a>
            </li>
          ) : (
            <li>
              <a
                onClick={() => {
                  this.props.changeCategory("Cloths");
                  this.props.fetchCategoryItems("Cloths");
                  this.setState({ activeCategory: "Cloths" });
                }}
              >
                Cloths
              </a>
            </li>
          )}

          {this.state.activeCategory === "Grocery" ? (
            <li>
              <a
                className="active"
                onClick={() => {
                  this.props.fetchCategoryItems("Grocery");
                  this.setState({ activeCategory: "Grocery" });
                }}
              >
                Grocery
              </a>
            </li>
          ) : (
            <li>
              <a
                onClick={() => {
                  this.props.changeCategory("Grocery");
                  this.props.fetchCategoryItems("Grocery");
                  this.setState({ activeCategory: "Grocery" });
                }}
              >
                Grocery
              </a>
            </li>
          )}

          {this.state.activeCategory === "Electronics" ? (
            <li>
              <a
                className="active"
                onClick={() => {
                  this.props.fetchCategoryItems("Electronics");
                  this.setState({ activeCategory: "Electronics" });
                }}
              >
                Electronics
              </a>
            </li>
          ) : (
            <li>
              <a
                onClick={() => {
                  this.props.changeCategory("Electronics");
                  this.props.fetchCategoryItems("Electronics");
                  this.setState({ activeCategory: "Electronics" });
                }}
              >
                Electronics
              </a>
            </li>
          )}
          <li className="search-bar">
            <i>
              <GoSearch />
            </i>
            <input
              className="search-input"
              type="text"
              onChange={this.onChange}
              placeholder=" Search..."
            ></input>
          </li>
        </ul>
      </div>
    );
  }
}

Navbar.propTypes = {
  fetchCategoryItems: PropTypes.func,
  fetchItems: PropTypes.func,
  searchItem: PropTypes.func,
  changeCategory: PropTypes.func,
}

export default Navbar;