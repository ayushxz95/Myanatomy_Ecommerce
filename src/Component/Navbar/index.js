import { connect } from "react-redux";
import { mapStateToProps, mapDispatchToProps } from "./props";
import Navbar from "./Navbar";

export default connect(mapStateToProps, mapDispatchToProps)(Navbar);