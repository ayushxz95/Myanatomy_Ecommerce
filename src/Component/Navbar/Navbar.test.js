import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Navbar from "./Navbar";

configure({ adapter: new Adapter() });


describe("Testing Navbar Component", () => {
  it("Matching Navbar's snapshot", () => {
    const component = shallow(
      <Navbar />
    );
    expect(component).toMatchSnapshot();
  });
});