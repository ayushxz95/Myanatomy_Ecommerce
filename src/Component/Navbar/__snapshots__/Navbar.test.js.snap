// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navbar Component Matching Navbar's snapshot 1`] = `
<div
  className="nav"
>
  <ul>
    <li
      className="hamberger-menu-container"
    >
      <Connect(ResNavbar) />
    </li>
    <li>
      <a
        className="active"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        onClick={[Function]}
      >
        Cloths
      </a>
    </li>
    <li>
      <a
        onClick={[Function]}
      >
        Grocery
      </a>
    </li>
    <li>
      <a
        onClick={[Function]}
      >
        Electronics
      </a>
    </li>
    <li
      className="search-bar"
    >
      <i>
        <GoSearch />
      </i>
      <input
        className="search-input"
        onChange={[Function]}
        placeholder=" Search..."
        type="text"
      />
    </li>
  </ul>
</div>
`;
